<!-- About Section -->
    <section id="research">
        <div class="container">
            <div class="row">
                <div class="col-lg-12 text-center">
                    <h2 class="section-heading">Research</h2>
                    <h3 class="section-subheading text-muted">The research in our laboratory is focused on the application of computational methods to develop a deeper understanding of genetic variation in diverse contexts. Modern experimental methods allow us to examine entire genomes with exquisite detail. Perhaps not surprisingly, staggering complexity is revealed as we look more closely at how genetic variation (both inherited and somatic) contributes to phenotypes. Modern genomic technologies necessitate efficient approaches for exploring, manipulating and comparing large genomic datasets. We develop such methods so that we and others may apply them to experiments investigating the impact of genetic variation on human disease, evolution, and somatic differentiation. Genome research is difficult - we strive to develop computational means that make it easier.</h3>
                </div>
            </div>
            <div class="row">
                <div class="col-lg-12">
                    <ul class="timeline">
                        <li>
                            <div class="timeline-image">
                                <img class="img-circle img-responsive" src="img/research/pedigree.square.jpg" alt="">
                            </div>
                            <div class="timeline-panel">
                                <div class="timeline-heading">
                                    <h3 class="subheading">Rare disease genetics</h3>
                                </div>
                                <div class="timeline-body">
                                    <p class="text-muted">We develop and apply new software for identifying causal genetic variants in studies of rare familial disease. The University of Utah has a long history of expertise in this area and we work closely with many clinical collaborators to solve rare disease. Our <a href="http://gemini.readthedocs.org/en/latest/">GEMINI</a> software is central to these efforts, and our laboratory collaborates with other members of the <a href="ucgd.genetics.utah.edu">USTAR Center for Genetic Discovery</a> to study familial disease among the large pedigrees in the <a href="http://healthsciences.utah.edu/utah-genome-project/">Utah Genome Project</a>.</p>
                                </div>
                            </div>
                        </li>
                        <li class="timeline-inverted">
                            <div class="timeline-image">
                                <img class="img-circle img-responsive" src="img/research/sv.png" alt="">
                            </div>
                            <div class="timeline-panel">
                                <div class="timeline-heading">
                                    <h3 class="subheading">Structural variation</h3>
                                </div>
                                <div class="timeline-body">
                                    <p class="text-muted">Human chromosomes harbor hundreds of structural differences including deletions, insertions, duplications, inversions, and translocations. Collectively, these differences are known as "structural variation" (or, "SV"). Any two humans differ by thousands of structural variants which vary greatly in size and phenotypic consequence. However, we are just beginning to understand the contribution of SV to evolution, development, and complex disease. Our laboratory continues to develop new methods such as <a href="https://github.com/arq5x/lumpy-sv">LUMPY</a> for detecting and understanding structural variation using modern DNA sequencing techniques.</p>
                                </div>
                            </div>
                        </li>
                        <li>
                            <div class="timeline-image">
                                <img class="img-circle img-responsive" src="img/research/cancer.png" alt="">
                            </div>
                            <div class="timeline-panel">
                                <div class="timeline-heading">
                                    <h3 class="subheading">Cancer genomics and genome evolution</h3>
                                </div>
                                <div class="timeline-body">
                                    <p class="text-muted">Massively parallel DNA sequencing has yielded detailed maps of clonal variation in human cancer, through an inference of clonal substructure by analysis of variant allele frequencies in bulk tumor cell populations and direct sequencing of single cells. Dynamic changes in clonal structure over time and under the selective pressure of treatment have been extensively studied in hematologic malignancies, but are less well characterized in solid cancers. Our understanding of the dynamics of clonal change and its role in therapeutic response and the emergence of resistance is in its infancy. However, deeper insight is accessible via significant advances in sequencing and new algorithms. We are developing new methods to identify genomic changes that are responsible for clonal evolution, chemoresistance, and relapse.</p>
                                </div>
                            </div>
                        </li>
                        <li class="timeline-inverted">
                            <div class="timeline-image">
                                <img class="img-circle img-responsive" src="img/research/code.cropped.png" alt="">
                            </div>
                            <div class="timeline-panel">
                                <div class="timeline-heading">
                                    <h3 class="subheading">Algorithm and Software Development</h3>
                                </div>
                                <div class="timeline-body">
                                    <p class="text-muted">Broadly speaking, the research in my laboratory marries genetics with genomics technologies, computer science, and machine learning techniques to develop new strategies for gaining insight into genome biology. We try to tackle challenging problems with practical importance to understanding genome variation in the context of human disease. We actively maintain a broad range of widely used tools for genome research including: <a href="http://bedtools.readthedocs.org/en/latest/">BEDTOOLS</a>, <a href="http://gemini.readthedocs.org/en/latest/">GEMINI</a>, <a href="https://github.com/arq5x/lumpy-sv">LUMPY</a>, <a href="https://github.com/brentp/vcfanno">VCFANNO</a>, <a href="https://github.com/brentp/pedagree">PEDAGREE</a>, and <a href="https://github.com/ryanlayer/gqt">GQT</p>.
                                </div>
                            </div>
                        </li>
                    </ul>
                </div>
            </div>
        </div>
    </section>